### PR TITLE
API review June 2022

### PR DESCRIPTION
**Pull request by msoeken**
This is to perform the [asynchronous API review meeting](https://github.com/microsoft/QuantumLibraries/wiki/Q%23-API-Design-Process#async-api-review-meeting) for June 2022. The issues are listed as part of the meeting notes. Please feel free to leave comments or edits.

### CODE REVIEW

File: Design/meetings/2022/api-design-2022-06.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+**Consensus**:
+
+---
+
+### Functions for smallest and largest representable fixed point
+
+**Proposal**: https://github.com/microsoft/QuantumLibraries/issues/594
+
+**Reviews**:
```
Comment (tcNickolas) [inline Design/meetings/2022/api-design-2022-06.md:28]:
```suggestion
**Reviews**:
* @tcNickolas, *approve*, left a comment
```

File: Design/meetings/2022/api-design-2022-06.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+**Consensus**:
+
+---
+
+### Conversion functions for signed integers
+
+**Proposal**: https://github.com/microsoft/QuantumLibraries/issues/595
+
+**Reviews**:
```
Comment (tcNickolas) [inline Design/meetings/2022/api-design-2022-06.md:39]:
```suggestion
**Reviews**:
* @tcNickolas, *approve*, left a comment
```

File: Design/meetings/2022/api-design-2022-06.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+- https://github.com/microsoft/QuantumLibraries/issues/594
+- https://github.com/microsoft/QuantumLibraries/issues/595
+
+## Discussion
+
+### Comparison operations
+
+**Proposal**: https://github.com/microsoft/QuantumLibraries/issues/593
+
+**Reviews**:
```
Comment (tcNickolas) [inline Design/meetings/2022/api-design-2022-06.md:17]:
```suggestion
**Reviews**:
* @tcNickolas, *approve*, left a comment
```

**Review by tcNickolas — APPROVED** [review]
I took a look at the proposals and left some comments in the respective issues. Overall looks good!

File: Design/meetings/2022/api-design-2022-06.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# Q# API Design Discussions / June 2022
+
+Reviewers (in order by username):
```
Comment (msoeken) [inline Design/meetings/2022/api-design-2022-06.md:3]:
```suggestion
Reviewers (in order by username): @tcNickolas
```

File: Design/meetings/2022/api-design-2022-06.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+**Consensus**:
+
+---
+
+### Conversion functions for signed integers
+
+**Proposal**: https://github.com/microsoft/QuantumLibraries/issues/595
+
+**Reviews**:
```
Comment (cgranade) [inline Design/meetings/2022/api-design-2022-06.md:41]:
```suggestion
**Reviews**:
* @cgranade: *comment*. We should coordinate this with the Q# language discussion around bit vectors in general. @swernli and @bettinaheim in particular may want to provide input here.
```

File: Design/meetings/2022/api-design-2022-06.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
+# Q# API Design Discussions / June 2022
+
+Reviewers (in order by username): @tcNickolas
```
Comment (msoeken) [inline Design/meetings/2022/api-design-2022-06.md:3]:
```suggestion
Reviewers (in order by username): @tcNickolas, @cgranade
```

File: Design/meetings/2022/api-design-2022-06.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+### Comparison operations
+
+**Proposal**: https://github.com/microsoft/QuantumLibraries/issues/593
+
+**Reviews**:
+- @cgranade, *comment*: Does the `action` input to `ApplyControlledOnLessThanFxP` need to be adjointable? Should there be a separate `Adj` variant? We may also want to describe quickly how this proposal works with the outstanding proposal for numerics refactoring all-up (https://github.com/microsoft/QuantumLibraries/issues/337).
+* @tcNickolas, *approve*, left a comment
+> Please add a bullet point including your alias, your review result (*approve*, *reject*, *comment*), and a comment (optional when result is *approve*).  Alternatively, add a line to the PR discussion incl. a reference to this issue.
+
+**Consensus**:
```
Comment (msoeken) [inline Design/meetings/2022/api-design-2022-06.md:22]:
```suggestion
**Consensus**: *approve* subject to comments to be addressed in the issue, discussion will be deferred to issue.
```

File: Design/meetings/2022/api-design-2022-06.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+### Functions for smallest and largest representable fixed point
+
+**Proposal**: https://github.com/microsoft/QuantumLibraries/issues/594
+
+**Reviews**: @cgranade, *approve*, same comment as above (https://github.com/microsoft/QuantumLibraries/issues/337).
+* @tcNickolas, *approve*, left a comment
+> Please add a bullet point including your alias, your review result (*approve*, *reject*, *comment*), and a comment (optional when result is *approve*).  Alternatively, add a line to the PR discussion incl. a reference to this issue.
+
+**Consensus**:
```
Comment (msoeken) [inline Design/meetings/2022/api-design-2022-06.md:34]:
```suggestion
**Consensus**: *approve*
```

File: Design/meetings/2022/api-design-2022-06.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+### Conversion functions for signed integers
+
+**Proposal**: https://github.com/microsoft/QuantumLibraries/issues/595
+
+**Reviews**:
+* @cgranade: *comment*. We should coordinate this with the Q# language discussion around bit vectors in general. @swernli and @bettinaheim in particular may want to provide input here.
+* @tcNickolas, *approve*, left a comment
+> Please add a bullet point including your alias, your review result (*approve*, *reject*, *comment*), and a comment (optional when result is *approve*).  Alternatively, add a line to the PR discussion incl. a reference to this issue.
+
+**Consensus**:
```
Comment (msoeken) [inline Design/meetings/2022/api-design-2022-06.md:47]:
```suggestion
**Consensus**: *approve* subject to comments to be addressed in the issue, discussion will be deferred to issue.
```

**Pull request merged**